src/Microsoft.Azure.SignalR.Emulator/Upstreams/HttpUpstreamTrigger.cs

```diff
@@ -0,0 +1,109 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Microsoft.Azure.SignalR.Common;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.SignalR.Emulator
+{
+    public class HttpUpstreamTrigger : IHttpUpstreamTrigger
+    {
+        private readonly IOptionsMonitor<UpstreamOptions> _optionsMonitor;
+        private readonly IHttpClientFactory _httpClientFactory;
+        private readonly ILogger _logger;
+
+        public HttpUpstreamTrigger(IOptionsMonitor<UpstreamOptions> optionsMonitor, IHttpClientFactory httpClientFactory, ILogger<HttpUpstreamTrigger> logger)
+        {
+            _optionsMonitor = optionsMonitor;
+            _httpClientFactory = httpClientFactory ?? throw new ArgumentNullException(nameof(httpClientFactory));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        public async Task<HttpResponseMessage> TriggerAsync(UpstreamContext context, IHttpUpstreamPropertiesFeature upstreamProperties, InvokeUpstreamParameters parameters, Action<HttpRequestMessage> configureRequest = null, CancellationToken token = default)
+        {
+            var request = new HttpRequestMessage(HttpMethod.Post, context.Uri);
+
+            request.Headers.Add(Constants.Headers.AsrsConnectionIdHeader, upstreamProperties.ConnectionId);
+            request.Headers.Add(Constants.Headers.AsrsHubNameHeader, parameters.Hub);
```

Comment:
Some constants have suffix "Header", but some don't, how about remove this suffix, since they are in Constants.Header class

Reply:
hm... the code is from runtime, I'd like to keep it as much un-modified as possible so that it can be easier for runtime and sdk to share some common protocol library later, as planned.